File: guilib-api/src/main/java/net/developertobi/guilib/api/gui/GuiHandler.java
package net.developertobi.guilib.api.gui;

import com.google.common.collect.Multimap;
import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;

public interface GuiHandler {

    Multimap<Player, GuiView> getInventories();

    void openStaticView(Player holder, Component title, Gui provider, boolean forceSyncOpening);

    default void cacheView(Player holder, Component title, Gui provider) {
        cacheView(holder, title, provider, false);
    }

    GuiView cacheView(Player holder, Component title, Gui provider, boolean staticGui);

    void updateCachedView(Player holder, String viewId);

    void clearCachedViews(Player holder);

    void removeCachedView(Player holder, GuiView view);

    GuiView getView(Player holder, String name);
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/inventory/GuiHandler.kt
package net.developertobi.inventorylib.api.inventory

import com.google.common.collect.Multimap
import net.kyori.adventure.text.Component
import org.bukkit.entity.Player

interface GuiHandler {

    val inventories: Multimap<Player, GuiView>

    fun openStaticView(holder: Player, title: Component, provider: Gui, forceSyncOpening: Boolean)

    fun cacheView(holder: Player, title: Component, provider: Gui)

    fun cacheView(
        holder: Player,
        title: Component,
        provider: Gui,
        staticInventory: Boolean
    ): GuiView?

    fun updateCachedView(holder: Player, viewId: String)

    fun clearCachedViews(holder: Player)

    fun removeCachedView(holder: Player, view: GuiView)

    fun getView(holder: Player, name: String): GuiView?

}

